docs/type.md

```diff
@@ -1,98 +1,197 @@
 ---
-title: Type!
+title: Types
 ---
 
-Types are the highlight of Reason! Here, you get a glimpse of why so many are excited about them.
+Types describe what kind of thing values are. Is this value a string, integer,
+or some complex structure? The type of that value will give a direct answer.
+Having a strict type system is a powerful tool that removes large classes of
+bugs and catches many others when compiling.
 
-This page briefly introduces the types syntax so that you can power through the subsequent sections without getting confused. More advanced topics on types can be found in the [More On Type](more-on-type.md) section.
+In Reason almost all types can be inferred. The compiler will figure out the
+types of everything in your program and ensure they make sense. This means you
+get the benefits of a strict type system without the drawbacks of writing down
+the type of every value.
 
-## Annotations
+## The Basics
+
+This is a [let binding](let-binding.md). The binding is named `count`, is of
+type `int`, and has a value of `42`. Its type was inferred, we did not
+explicitly write down that it is an `int`.
 
-This let-binding doesn't contain any written type:
+_Everything_ in Reason has a type, even if you do not write it down.
 
-<!--DOCUSAURUS_CODE_TABS-->
-<!--Reason-->
 ```reason
-let score = 10;
+let count = 42;
 ```
-<!--Output-->
-```js
-var score = 10;
+
+Types can be explicitly added with an annotation:
+
+```reason
+let count: int = 42;
 ```
-<!--END_DOCUSAURUS_CODE_TABS-->
 
-Reason deducted that `score` is an `int`, judging by the value `10`. This is called **inference**.
+Because `count` has a type the compiler knows what we are and are not allowed
+to do with its value:
 
-But types can also be explicitly written down by choice:
+```reason
+/* Allowed: addition */
+let nextCount = count + 1;
+
+/* Error: count is not a list */
+let x = List.map(fn, count);
+```
+
+## Annotations
+
+Type annotations can appear almost anywhere. They are not often necessary due to
```

Comment:
Worth mentioning that they need to reside within parenthesis, or between commas of lists, tuples, and function arguments/parameters.

Reply:
Alternatively, you could suggest that they should always be placed with surrounding parenthesis and refmt will remove them where they weren't actually necessary. (If anyone has used Flow they probably would get this).